src/java/org/apache/cassandra/db/marshal/UserType.java

```diff
@@ -542,7 +542,7 @@ public List<ByteBuffer> filterSortAndValidateElements(List<ByteBuffer> buffers)
             ByteBuffer buffer = buffers.get(i);
             if (buffer == null)
                 continue;
-            if (!isMultiCell() && buffer == ByteBufferUtil.UNSET_BYTE_BUFFER)
```

Comment:
UserTypesTest.testUDTWithUnsetValues would fail without this (while with other changes in place). That test has to fail when some field in UDT is set to be empty. Previously, in order to throw, isMultiCell had to be false and it was because we were coming into this method with a type which was frozen and frozen is not multi-cell. When we unfroze, it became multicell again, so `if` would evaluate to `false`, so it would not throw, so the test would fail.